### PR TITLE
code annotation for Editor directory

### PR DESCRIPTION
**Pull request by MrRowey**
I've done what i think is right any that i'm unsure of i've left as any

- [x] AMPlatoonHelperFunctions
- [x] BaseManagerBuildConditions
- [x] EconomyBuildCondions
- [x] InstantBuildConditions
- [x] MarkerBuildConditions
- [x] MiscBuildConditions
- [x] OtherArmyUnitCountBuildConditions
- [x] PlattonCountBuildContions
- [x] SorianBuildContions
- [x] SorianInstantBuildConditions
- [x] ThreatBuildConditions
- [x] UnitCountBuildConditions

### CODE REVIEW

File: lua/editor/AMPlatoonHelperFunctions.lua [hunk 1]
```diff
@@ -57,44 +43,34 @@ function AMUnlockPlatoon(brain, platoon)
...
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
--- function: AMUnlockPlatoonTimer = BuildCallback   doc = "Please work function docs."
--- 
--- parameter 0: string	brain		= "default_brain"		
--- parameter 1: string	platoon  	= "default_platoon"		doc = "docs for param1"
--- parameter 2: int      duration        = "120"
---
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+--- AMUnlockPlatoonTimer = BuildCallback
+---@param brain string
```
Comment (Garanas) [inline lua/editor/AMPlatoonHelperFunctions.lua:47]: The `brain` is usually going to be of type `AIBrain`. A platoon is usually `platoon` of type `Platoon`, unless it has `name` in it. You can see it if they manage to write `platoon.PlatoonData`, then it is a platoon.

File: lua/editor/BaseManagerBuildConditions.lua [hunk 2]
```diff
@@ -3,16 +3,15 @@
...
 
 local AIUtils = import('/lua/ai/aiutilities.lua')
-local ScenarioFramework = import('/lua/scenarioframework.lua')
-local ScenarioUtils = import('/lua/sim/ScenarioUtilities.lua')
 
--- function: NeedAnyStructure = BuildCondition    doc = "Please work function docs."
--- parameter 0: string    aiBrain        = "default_brain"
--- parameter 1: string    baseName = "base_name"
+---NeedAnyStructure = BuildCondition
+---@param aiBrain string
```
Comment (Garanas) [inline lua/editor/BaseManagerBuildConditions.lua:12]: Definitely an AI Brain, looking at how the code is used

File: lua/editor/EconomyBuildConditions.lua [hunk 3]
```diff
@@ -9,18 +9,12 @@
...
--- function: GreaterThanEconStorageRatio = BuildCondition	doc = "Please work function docs."
---
--- parameter 0: string	aiBrain		= "default_brain"
--- parameter 1: float	mStorageRatio	= 0.0				doc = "docs for param1"
--- parameter 2: float	eStorageRatio	= 0.0				doc = "param2 docs"
---
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+---GreaterThanEconStorageRatio = BuildCondition
+---@param aiBrain string
```
Comment (Garanas) [inline lua/editor/EconomyBuildConditions.lua:14]: Again, def. an `AIBrain` instead of a string!

File: lua/editor/InstantBuildConditions.lua [hunk 4]
```diff
@@ -10,12 +10,8 @@ local ScenarioFramework = import('/lua/scenarioframework.lua')
 local ScenarioUtils = import('/lua/sim/ScenarioUtilities.lua')
 local Utils = import('/lua/utilities.lua')
 
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
--- function: PreBuiltBase = BuildCondition  doc = "Please work function docs."
---
--- parameter 0: string   aiBrain         = "default_brain"
---
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+---@param aiBrain string
```
Comment (Garanas) [inline lua/editor/InstantBuildConditions.lua:13]: Here too - it uses fields that a string doesn't have. Def. of type AIBrain!

**Review by Garanas — CHANGES_REQUESTED** [review]
Usually, `aibrain` or `brain` is of type `AIBrain`, and `platoon` is of type `Platoon`. You can inspect this by looking what values / functions are expected of the parameter (in the body of the function). It appears the old documentation was a bit off / difficult to understand, as I think it is referring to how the parameter is constructed and not what it is.

@speed2CZ could you scan this pull request too?

**Comment by Garanas** [comment]
Better - you can resolve suggestions that you've applied.

**Comment by MrRowey** [comment]
@Garanas all 12 files are done thier a few param that im unsure of ans same gose for a few unknown returns

File: lua/editor/BaseManagerBuildConditions.lua [hunk 5]
```diff
@@ -3,16 +3,15 @@
 -- Author(s): Dru Staltman, John Comes
 -- Summary  : Generic AI Platoon Build Conditions
 --           Build conditions always return true or false
--- Copyright � 2005 Gas Powered Games, Inc.  All rights reserved.
+-- Copyright � 2005 Gas Powered Games, Inc.  All rights reserved.
```
Comment (Garanas) [inline lua/editor/BaseManagerBuildConditions.lua:6]: Can you fix the copyright symbols?

File: lua/editor/EconomyBuildConditions.lua (hunk 3 above)
Comment (Garanas) [reply lua/editor/EconomyBuildConditions.lua:14]: It is still a string here, it should be AIBrain

File: lua/editor/InstantBuildConditions.lua [hunk 6]
```diff
@@ -82,15 +62,11 @@ function HaveGreaterThanUnitsWithCategory(aiBrain, numReq, category, idleReq)
...
---
--- parameter 0: string	aiBrain		= "default_brain"
--- parameter 1: int	numReq          = 0				doc = "docs for param1"
--- parameter 2: expr   category        = categories.ALLUNITS		doc = "param2 docs"
--- parameter 3: expr   idleReq       = false         doc = "docs for param3"
---
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+---@param aiBrain AIBrain
+---@param numReq integer
+---@param category EntityCategory
```
Comment (Garanas) [inline lua/editor/InstantBuildConditions.lua:67]: @Hdt80bro is this the correct type for entity categories?

Comment (Hdt80bro) [reply lua/editor/InstantBuildConditions.lua:67]: Yup! `EntityCategory` represents the underlying C Object that handles addition, multiplication, and subtraction of categories.
I aliased `UnparsedCategory` to `string` for unparsed categories too, just to make that more clear to the user.

**Pull request merged**